src/compiler/parser.rs

```diff
@@ -613,6 +651,26 @@ impl ParserType {
         false
     }
 
+    fn parse_expected_token(
+        &self,
+        t: SyntaxKind,
+        diagnostic_message: Option<&DiagnosticMessage>,
+        args: Option<Vec<String>>,
+    ) -> Node {
+        self.parse_optional_token(t).unwrap_or_else(|| {
+            enum_unwrapped!(
```

Comment:
Could add the same ergonomic eg `.as_template_literal_like_node()` methods to `MissingNode` as I've been doing for `Node`/`Type`